lib/internal/promisify.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+module.exports = promisify;
+
+function promisify(fn, single) {
+  Object.defineProperty(inner, 'name', {
+    value: fn.name + 'Promisified'
+  });
+  return inner;
+
+  function inner() {
+    if (typeof arguments[arguments.length - 1] === 'function') {
+      return fn.apply(this, arguments);
```

Comment:
```javascript
Array.prototype.push.call(parameters, function(error, data) {
        if (error) {
          reject(error)
        } else resolve(data)
      })
```

this part will be particularly inefficient because you pass `arguments` to external function.